pkg/sql/sqlinstance/instancestorage/instancereader.go

```diff
@@ -225,8 +264,9 @@ func (r *Reader) GetAllInstances(
 	return sqlInstances, nil
 }
 
-func (r *Reader) getAllLiveInstances(ctx context.Context) ([]instancerow, error) {
-	rows := r.getAllInstanceRows()
+func (r *Reader) getAllUniqueLiveInstances(
```

Comment:
uber-nit: I think I'd take this off of the `*Reader` and make it a free standing function which takes a `sqlliveness.Reader`. I'd probably call it `selectDistinctLiveRows(context.Context, sqlliveness.Reader, []instanceRow) ([]instancerow, error)`

Reply:
Note also that the function will modify the slice in place.